contrib/platform/src/com/sun/jna/platform/win32/Kernel32Util.java

```diff
@@ -424,70 +427,28 @@ public static String getEnvironmentVariable(String name) {
             return null;
         }
 
-        Map<String,String>  vars=new TreeMap<String,String>();
-        boolean             asWideChars=isWideCharEnvironmentStringBlock(lpszEnvironmentBlock, offset);
-        long                stepFactor=asWideChars ? 2L : 1L;
-        for (long    curOffset=offset; ; ) {
-            String  nvp=readEnvironmentStringBlockEntry(lpszEnvironmentBlock, curOffset, asWideChars);
-            int     len=nvp.length();
-            if (len == 0) { // found the ending '\0'
+        Map<String,String> vars = new TreeMap<String,String>();
+        for (long curOffset = offset; ;) {
+            String nvp = NativeWin32String.getString(lpszEnvironmentBlock, curOffset);
+            if (nvp.isEmpty()) { // found the ending '\0'
                 break;
             }
 
-            int pos=nvp.indexOf('=');
+            int pos = nvp.indexOf('=');
             if (pos < 0) {
                 throw new IllegalArgumentException("Missing variable value separator in " + nvp);
             }
 
-            String  name=nvp.substring(0, pos), value=nvp.substring(pos + 1);
+            String name = nvp.substring(0, pos);
+            String value = nvp.substring(pos + 1);
             vars.put(name, value);
 
-            curOffset += (len + 1 /* skip the ending '\0' */) * stepFactor;
+            curOffset += nvp.length() + 1;
         }
 
         return vars;
     }
 
-    /**
-     * @param lpszEnvironmentBlock The environment block as received from the
-     * <A HREF="https://msdn.microsoft.com/en-us/library/windows/desktop/ms683187(v=vs.85).aspx">GetEnvironmentStrings</A>
-     * function
-     * @param offset Offset within the block to look for the entry
-     * @param asWideChars If {@code true} then the block contains {@code wchar_t}
-     * instead of &quot;plain old&quot; {@code char}s
-     * @return A {@link String} containing the <code>name=value</code> pair or
-     * empty if reached end of block
-     * @see #isWideCharEnvironmentStringBlock
-     * @see #findEnvironmentStringBlockEntryEnd
-     */
-    public static String readEnvironmentStringBlockEntry(Pointer lpszEnvironmentBlock, long offset, boolean asWideChars) {
```

Comment:
Where did this whole method go?

Reply:
Was intended to mark as deprecated, but somehow got removed during my squashing. Will be restored on next push.